composer/loggers/wandb_logger.py

```diff
@@ -335,4 +354,36 @@ def _convert_to_wandb_image(image: Union[np.ndarray, torch.Tensor], channels_las
     if not channels_last:
         assert isinstance(image, np.ndarray)
         image = image.transpose(1, 2, 0)
-    return image
+    return np.asarray(image)
+
+
+def _convert_to_wandb_mask(mask: Union[np.ndarray, torch.Tensor], channels_last: bool) -> np.ndarray:
+    mask = _convert_to_wandb_image(mask, channels_last)
+    mask = mask.squeeze()
+    if mask.ndim != 2:
+        raise ValueError(f'Mask must be a 2D array, but instead got array of shape: {mask.shape}')
+    return mask
+
+
+def _preprocess_mask_data(masks: Dict[str, Union[np.ndarray, torch.Tensor, Sequence[Union[np.ndarray, torch.Tensor]]]],
+                          channels_last: bool) -> Dict[str, np.ndarray]:
+    preprocesssed_masks = {}
+    for mask_name, mask_data in masks.items():
+        if not isinstance(mask_data, Sequence):
+            mask_data = mask_data.squeeze()
+            if mask_data.ndim == 2:
+                mask_data = [mask_data]
+        preprocesssed_masks[mask_name] = np.stack([_convert_to_wandb_mask(mask, channels_last) for mask in mask_data])
+    return preprocesssed_masks
+
+
+def _create_wandb_masks_generator(masks: Dict[str, Union[np.ndarray, torch.Tensor, Sequence[Union[np.ndarray,
```

Comment:
It should be good because it converts Tensors to numpy arrays in `_convert_to_wandb_image` and everything has to go through that function.

Reply:
The masks dict does not go through `_convert_to_wandb_image` I think? So I think right now if you actually passed a torch tensor as the masks to `WandBLogger.log_images` I think it would crash. It is mostly a non issue because your code is the only caller of this function, but it would be good to get the types right here.

Reply:
ah nevermind, i see it now